README.md

```diff
@@ -81,24 +87,61 @@ Wrappers (see [list](https://github.com/gyyang/neurogym/blob/master/docs/wrapper
 
-All options in the configuration code have default values. Rather than changing the values in the code, you can use a TOML file to override the defaults. To do that, just copy the [template configuration](docs/examples/config.toml), change the values of any options that you need to customise, and save it somewhere convenient. You can pass that file to the `Config` class to create a custom configuration vobject:
+🧪 **Beta Feature** — The configuration system is **optional** and currently **under development**. You can still instantiate environments, agents, and wrappers with direct parameters.
+It is only used in a small portion of the codebase and is not required for typical usage.
+See the [`demo.ipynb`](docs/examples/demo.ipynb) notebook for the only current example of this system in action.
+
+NeuroGym includes a flexible configuration mechanism using [`Pydantic Settings`](https://docs.pydantic.dev/latest/concepts/pydantic_settings/), allowing configuration via TOML files, Python objects, or plain dictionaries.
+
+#### 1. From a TOML file
+
+Create a `config.toml` file (see [template](docs/examples/config.toml)) and load it:
 
 ```python
 from neurogym import Config
-config = Config('<path-to-TOML-file>')
+config = Config('path/to/config.toml')
 ```
 
-The resulting object can be used on its own, or it can be passed to a wrapper (such as a monitor). It is also possible to pass the path to the TOML configuration file directly to the monitor when instantiating it:
+You can then pass this config to any component that supports it:
 
 ```python
 from neurogym.wrappers import monitor
 env = gym.make('GoNogo-v0')
-env = monitor.Monitor(env, config="config.toml") # You can also pass a `Config` object instead of a path.
+env = monitor.Monitor(env, config=config)
+```
+
+Or directly pass the path:
+
+```python
+env = monitor.Monitor(env, config='path/to/config.toml')
+```
+
+#### 2. With Python class
+
+```python
+from neurogym import Config
+config = Config(
+    local_dir="logs/",
+    env={"name": "GoNogo-v0"},
+    monitor={"name": "MyMonitor"}
+)
 ```
 
-Please refer to the [template configuration](docs/examples/config.toml) for more details about the available options. The goal is to implement as many configurable options as possible in this configuration system in order to allow for environments, agents, experiments and so on to be instantiated just by passing a configuration file.
+#### 3. With a dictionary
+
+```python
+from neurogym import Config
+config_dict = {
+    "env": {"name": "GoNogo-v0"},
+    "monitor": {
+        "name": "MyMonitor",
+        "plot": {"trigger": "steps", "value": 500, "create": True}
```

Comment:
Should be  `step` rather than `steps`

`"plot": {"trigger": "step", "value": 500, "create": True}`